packages/twenty-front/src/modules/object-record/record-table/empty-state/components/RecordTableEmptyStateDisplay.tsx

```diff
@@ -11,41 +11,49 @@ import {
   IconComponent,
 } from 'twenty-ui';
 
+type RecordTableEmptyStateDisplayButtonComponentProps = {
+  buttonComponent?: React.ReactNode;
+};
+
+type RecordTableEmptyStateDisplayButtonProps = {
+  ButtonIcon: IconComponent;
+  buttonTitle: string;
+  onClick: () => void;
+};
+
 type RecordTableEmptyStateDisplayProps = {
   animatedPlaceholderType: AnimatedPlaceholderType;
   title: string;
   subTitle: string;
-  Icon: IconComponent;
-  buttonTitle: string;
-  onClick: () => void;
-};
+} & (
+  | RecordTableEmptyStateDisplayButtonComponentProps
+  | RecordTableEmptyStateDisplayButtonProps
+);
 
-export const RecordTableEmptyStateDisplay = ({
-  Icon,
-  animatedPlaceholderType,
-  buttonTitle,
-  onClick,
-  subTitle,
-  title,
-}: RecordTableEmptyStateDisplayProps) => {
+export const RecordTableEmptyStateDisplay = (
+  props: RecordTableEmptyStateDisplayProps,
```

Comment:
Let's keep props destructuring no ?

Reply:
I don't do props destructuring here because some of them are conditional, `buttonComponent` and `ButtonIcon` can't be provided at the same time